src/boot/efi/meson.build

```diff
@@ -136,6 +136,45 @@ if have_gnu_efi
         endif
 endif
 
+sbat_substs = configuration_data()
+sbat_substs.set('sbat_project', meson.project_name())
+sbat_substs.set('PROJECT_VERSION', substs.get('PROJECT_VERSION'))
+sbat_substs.set('PROJECT_URL', substs.get('PROJECT_URL'))
+sbat_substs.set('sbat_distro_generation', get_option('sbat_distro_generation'))
+sbatvars = [['sbat_distro', 'ID'],
+            ['sbat_distro_summary', 'NAME'],
+            ['sbat_distro_url', 'BUG_REPORT_URL']]
+foreach sbatvar : sbatvars
+        value = get_option(sbatvar[0])
+        if value == ''
+                value = run_command('sed', '-n', 's/^' + sbatvar[1] + '=//p', '/etc/os-release').stdout().strip()
+                value = value.strip('"')
+        endif
+        if value != ''
+                sbat_substs.set(sbatvar[0], value)
+        endif
+endforeach
+
+sbatvars = [['sbat_distro_pkgname', meson.project_name()],
+            ['sbat_distro_version', get_option('version-tag')]]
+foreach sbatvar : sbatvars
+       value = get_option(sbatvar[0])
+       if value == ''
+               value = sbatvar[1]
+       endif
+       sbat_substs.set(sbatvar[0], value)
+endforeach
+if sbat_substs.get('sbat_distro_version') == ''
+       sbat_substs.set('sbat_distro_version', substs.get('PROJECT_VERSION'))
+endif
+
+
+if efi_arch == 'x86_64' and sbat_substs.get('sbat_distro', '') != ''
```

Comment:
why the conditionalization on the arch? it appears to me that we should always generate this if people specify one of these sbat things on the meson cmdline, regardless of the arch

Reply:
Because i have not yet tested the shims on other arches that they link and boot correctly. We can generate this for all arches, but at the moment I can only verify this on x86_64 & arm64. There are shims for armhf i386. I don't think anybody have built or signed shims for riscv64 yet.

Ideally i want this on all EFI arches, but I am uncertain about untested .sbat sections. If you are ok with merging untested .sbat sections for a given arch, i can lift this.